ext/liquid_c/parser.c

```diff
@@ -130,6 +130,17 @@ static VALUE parse_variable(parser_t *p)
     return rb_class_new_instance(4, args, cLiquidVariableLookup);
 }
 
+bool will_parse_constant_expression_next(parser_t *p)
+{
+    switch (p->cur.type) {
+        case TOKEN_IDENTIFIER:
+        case TOKEN_OPEN_SQUARE:
```

Comment:
I wasn't too concerned about this function identifying all constants, since it is only used as part of an optimization for calling filters with constant arguments.  I did add support for constant ranges in the corresponding `try_parse_constant_expression` function in the follow-up PR (https://github.com/Shopify/liquid-c/pull/60), so I could backport that case to this PR.

Reply:
> Also potentially TOKEN_DOTDOT for completeness though it doesn't look necessary based on the places this is called.

`TOKEN_DOTDOT` isn't a valid token for starting a constant expression, so handling it with the default case is fine.